docs/docs/crafty.md

```diff
@@ -3,4 +3,184 @@ id: crafty
 title: Crafty
 ---
 
-Crafty
\ No newline at end of file
+Have you ever wondered what may happen if you combined an Aragon token and fire? A fire-spewing eagle? Burnt chicken? What if Augur's oracles were augmented with Decentraland's tiles, would they start predicting the real estate market? Now, you no longer need to ponder at these vital questions: the Ethereum community will answer them for you!
+
+In [Crafty](https://crafty.zeppelin.solutions), users can create new ERC20 tokens, with a few added goodies (following [EIP-1046](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1046.md)): a picture, and a short description. This allows for each token to have that extra bit of personality that makes something unique. But that's not all: the only way to get these tokens is by crafting them, which requires ingredients (any ERC20!) to be spent. And since the craftable tokens themselves can also be used as ingredients, the posibilities are endless!
+
+## Project setup
+
+This guide will show you the process of making the contracts in the Crafty game upgradeable by using ZeppelinOS, which we'll install globally to have it available on all our projects.
+
+```sh
+npm install --global zos
+```
+
+All snippets here were extracted from the public [Crafty repository](https://github.com/zeppelinos/crafty).
+
+## Contracts
+
+Crafty works with only two contracts:
+* [`Crafty`](https://github.com/zeppelinos/crafty/blob/master/contracts/Crafty.sol) itself, containing the game logic (how new recipes are added, how crafting takes place, etc.) plus some role-based access control ([RBAC](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/zos-release/contracts/ownership/rbac/RBAC.sol), powered by [OpenZeppelin](https://openzeppelin.org/)), for admin tasks, such as deletion of tokens.
+* The [`CraftableToken`](https://github.com/zeppelinos/crafty/blob/master/contracts/CraftableToken.sol)s, which are a simple extension of a mintable ERC20, holding token creator data, a URI for metadata, and a list of required ingredients, along with the amounts for each.
+
+To better see how these two contracts interact, take a look at `Crafty`'s `craft` method, where a user provides the address of a `CraftableToken` to craft, and `Crafty` makes sure the required ingredients are consumed before minting it.
+
+```
+function craft(CraftableToken _craftable) public {
+  address player = msg.sender;
+
+  uint256 totalSteps = _craftable.getTotalRecipeSteps();
+  for (uint i = 0; i < totalSteps; ++i) {
+    ERC20 ingredient;
+    uint256 amountNeeded;
+    (ingredient, amountNeeded) = _craftable.getRecipeStep(i);
+
+    ingredient.transferFrom(player, address(this), amountNeeded);
+  }
+
+  // Issue the crafted token - for ingredient-less tokens, a higher number of
+  // tokens are minted.
```

Comment:
Ingredient-less (also called basic) tokens will probably be used by multiple craftables, and they are indivisible (i.e. decimals is 0), so I mint many of those on each call so that users don't need to do it multiple times. I should probably remove that from the docs though, it adds no value and may cause confusion.